### PR TITLE
RATIS-1693. Remove the use of the thirdparty Timer.

### PR DESCRIPTION
**Pull request by szetszwo**
See https://issues.apache.org/jira/browse/RATIS-1693

### CODE REVIEW

File: ratis-metrics/src/main/java/org/apache/ratis/metrics/impl/TimekeeperImpl.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ratis.metrics.impl;
+
+import org.apache.ratis.metrics.Timekeeper;
+import org.apache.ratis.thirdparty.com.codahale.metrics.Timer;
+
+public class TimekeeperImpl implements Timekeeper {
```
Comment (codings-dan) [inline ratis-metrics/src/main/java/org/apache/ratis/metrics/impl/TimekeeperImpl.java:23]: Whether there are other implementations of the Timekeeper interface in the future, if so, I suggest naming it as DefaultTimekeeperImpl

**Review by codings-dan — COMMENTED** [review]
@szetszwo Thanks for working on this! Overall lgtm, left some comments for reference.

File: ratis-metrics/src/main/java/org/apache/ratis/metrics/Timekeeper.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ratis.metrics;
+
+import org.apache.ratis.util.UncheckedAutoCloseable;
+
+import java.util.Optional;
+
```
[the remark this answers is not in the document]
Comment (szetszwo) [reply ratis-metrics/src/main/java/org/apache/ratis/metrics/Timekeeper.java:23]: @codings-dan , thanks for the reminder!  I kept forgetting to add the @ FunctionalInterface annotation.

File: ratis-metrics/src/main/java/org/apache/ratis/metrics/impl/TimekeeperImpl.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  static class ContexImpl implements Context {
```
[the remark this answers is not in the document]
Comment (szetszwo) [reply ratis-metrics/src/main/java/org/apache/ratis/metrics/impl/TimekeeperImpl.java:24]: Good catch!   We actually can use lambda instead.

File: ratis-metrics/src/main/java/org/apache/ratis/metrics/impl/TimekeeperImpl.java (hunk 1 above)
Comment (szetszwo) [reply ratis-metrics/src/main/java/org/apache/ratis/metrics/impl/TimekeeperImpl.java:23]: That's  a good idea.  Let's call it `DefaultTimekeeperImpl`.

**Review by codings-dan — APPROVED** [review]
The change looks good, +1

**Pull request merged**